pandas/tests/io/test_s3.py

```diff
@@ -23,3 +25,23 @@ def test_streaming_s3_objects():
     for el in data:
         body = StreamingBody(BytesIO(el), content_length=len(el))
         read_csv(body)
+
+
+@tm.network
+@pytest.mark.slow
+def test_read_s3_public():
+    # ensure we can read from a public bucket with credentials
+    pytest.importorskip("s3fs")
```

Comment:
Can you use the `skip_if_no` decorator from pandas/util/_test_decorators here instead?